concent_api/api-integration-test.py

```diff
@@ -0,0 +1,208 @@
+    assert 'type' not in message.__slots__
+    fields = ['type'] + HEADER_FIELDS + sorted(set(message.__slots__) - set(HEADER_FIELDS) - PRIVATE_FIELDS)
+    values = [
+        type(message).__name__ if field == 'type' else
+        getattr(message, field)
+        for field in fields
+    ]
+
+    for field, value in zip(fields, values):
+        if isinstance(value, bytes):
+            try:
+                nested_message = load(
+                    value,
+                    CONCENT_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY,
+                )
+            except AttributeError:
+                # FIXME: golem-messages provides no reliable way to discern invalid messages from other AttributeErrors.
+                print('{}{:30} = <BINARY DATA>'.format(' ' * indent, field))
+            else:
+                print('{}{:30} ='.format(' ' * indent, field))
+                print_golem_message(nested_message, indent = indent + 4)
+        else:
+            print('{}{:30} = {}'.format(' ' * indent, field, value))
+
+
+def api_request(host, endpoint, private_key = None, data = None, headers = None, ):
+    assert all(value not in ['', None] for value in [endpoint, host, headers])
+    import ipdb; ipdb.set_trace()
+    url = "{}/api/v1/{}/".format(host, endpoint)
+
+    if data is None:
+        print('RECEIVE ({})'.format(url))
+    else:
+        print('SEND ({})'.format(url))
+        print('MESSAGE:')
+        print_golem_message(data)
+
+        data = dump(
+            data,
+            private_key,
+            CONCENT_PUBLIC_KEY,
+        )
+
+    if data is None:
+        response = requests.post("{}".format(url), headers = headers)
+    else:
+        response = requests.post("{}".format(url), headers = headers, data = data)
+
+    if len(response.content) != 0:
+        decoded_response = load(
+            response.content,
+            CONCENT_PRIVATE_KEY,
+            CONCENT_PUBLIC_KEY,
+        )
+        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+        print('MESSAGE:')
+        print_golem_message(decoded_response)
+    else:
+        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+        if response.text not in ['', None]:
+            print('RAW RESPONSE: {}'.format(response.text))
+    print()
+
+
+def parse_command_line(command_line):
+    if len(command_line) <= 1:
+        sys.exit('Not enough arguments')
+
+    if len(command_line) >= 3:
+        sys.exit('Too many arguments')
+
+    cluster_url = command_line[1]
+    return cluster_url
+
+
+def force_report_computed_task(task_id, provider_private_key, provider_public_key, requestor_private_key, requestor_public_key, current_time):
+
+    compute_task_def = ComputeTaskDef()
+    compute_task_def['task_id'] = task_id
+    compute_task_def['deadline'] = current_time + 60
+    task_to_compute = TaskToCompute(
+        timestamp = current_time - 3540,
```

Comment:
How about putting this value 3540 in settings ? Or maybe it would never need to be changed, then its ok.